### DISCUSSION TITLE
Unload Textures

### DISCUSSION THREAD

**Discussion by alexeyKrIgo** [AlmasB/FXGL#1270] [Q&A] [answered]
If I have levels like lv 1 and lv 2 maps and each one of them uses different textures, how can I get rid of lv 1 textures when I enter the lv 2 map. If I understood correctly the FXGL.getAssetLoader().loadTexture() loads an asset, but will it stay in the memory when I pass to the next lvls and won't use the previous lvls asset anymore?

**Comment by AlmasB** [comment, marked as answer]
If you wish to free resources related to images, then you can:
1. `FXGL.getAssetLoader().clearCache()`
2. `texture.dispose()`

The memory associated with the image should then be freed on the next GC call.
